kyo-prelude/shared/src/main/scala/kyo/Aspect.scala

```diff
@@ -4,180 +4,223 @@ import Aspect.*
 
 /** Represents an aspect in the Kyo effect system.
   *
-  * An aspect allows for the modification of behavior across multiple points in a program without changing the code of the affected points
-  * directly. It provides a way to "weave" additional behavior into the existing code.
+  * An aspect provides a way to modify or intercept behavior across multiple points in a program without directly changing the affected
+  * code. It works by allowing users to provide implementations for abstract operations at runtime, similar to dependency injection but with
+  * more powerful composition capabilities.
   *
-  * @tparam A
-  *   The input type of the aspect
-  * @tparam B
-  *   The output type of the aspect
+  * Aspects are created using `Aspect.init` and are typically stored as vals at module level. Once initialized, an aspect can be used to
+  * wrap computations that need to be modified, and its behavior can be customized using the `let` method to provide specific
+  * implementations within a given scope. This pattern allows for clean separation between the definition of interceptable operations and
+  * their actual implementations.
+  *
+  * Aspects support multi-shot continuations, meaning that cut implementations can invoke the continuation function multiple times or not at
+  * all. This enables powerful control flow modifications like retry logic, fallback behavior, or conditional execution.
+  *
+  * Internally, aspects function as a form of reified ArrowEffect that can be stored, passed around, and modified at runtime. They maintain
+  * state through a `Local` map of active implementations, allowing them to be dynamically activated and deactivated through operations like
+  * `let` and `sandbox`.
+  *
+  * @tparam Input
+  *   The input type constructor - what values the aspect can process
+  * @tparam Output
+  *   The output type constructor - what results the aspect can produce
   * @tparam S
-  *   The effect type of the aspect
+  *   The effect type - what effects the aspect can perform
   */
-final class Aspect[A, B, S] private[kyo] (default: Cut[A, B, S])(using Frame) extends Cut[A, B, S]:
+final class Aspect[Input[_], Output[_], S] private[kyo] (
```

Comment:
I want to introduce an aspect to `Parse.read` to provide a `Parse.spaced` combinator that automatically consumes  whitespaces between parsers. It's not possible with the current implementation because `Parse.read` has a generic `A` type parameter.

This change uses an approach similar to `ArrowEffect` with two `Input[_]` and `Output[_]` type parameters to support aspects in generic contexts like `Read.parse`.